repose-aggregator/commons/commons-test/src/main/scala/org/openrepose/commons/test/ConfigValidator.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.commons.test
+
+import java.io.ByteArrayInputStream
+import javax.xml.transform.Source
+import javax.xml.transform.stream.StreamSource
+import javax.xml.validation.{SchemaFactory, Validator}
+
+class ConfigValidator(validator: Validator) {
+  def validateConfigFile(fileName: String): Unit = {
+    validator.validate(new StreamSource(classOf[ConfigValidator].getResourceAsStream(fileName)))
+  }
+
+  def validateConfigString(config: String): Unit = {
+    validator.validate(new StreamSource(new ByteArrayInputStream(config.getBytes)))
+  }
+}
+
+object ConfigValidator {
+  def apply(schemaFileName: String): ConfigValidator = apply(Array(schemaFileName))
+
+  def apply(schemaFileNames: Array[String]): ConfigValidator = {
```

Comment:
This works, but I am curious: why not use a var arg here instead? That is:
```
def apply(schemaFileNames: String*): ConfigValidator = {
  ...
}
```

I would think var args are generally easier to use. A user can either pass any Seq (`apply(Array("foo", "bar"): _*)`) or just their arguments (`apply("foo", "bar")`). Also, you can still use the bound variable as an array (it's technically typed as a Seq, specifically a WrappedArray).